src/contexts/TokenData.jsx

```diff
@@ -237,87 +237,84 @@ const getTopTokens = async (client, blockClient, nativeCurrencyPrice, nativeCurr
+
+        // catch the case where token wasnt in top list in previous days
+        if (!oneDayHistory) {
+          let oneDayResult = await client.query({
+            query: TOKEN_DATA(token.id, oneDayBlock),
+          });
+          oneDayHistory = oneDayResult.data.tokens[0];
+        }
+        if (!twoDayHistory) {
+          let twoDayResult = await client.query({
+            query: TOKEN_DATA(token.id, twoDayBlock),
           });
+          twoDayHistory = twoDayResult.data.tokens[0];
+        }
+
+        // calculate percentage changes and daily changes
+        const [oneDayVolumeUSD, volumeChangeUSD] = get2DayPercentChange(
+          data.tradeVolumeUSD,
+          oneDayHistory?.tradeVolumeUSD ?? 0,
+          twoDayHistory?.tradeVolumeUSD ?? 0,
+        );
+        const [oneDayTxns, txnChange] = get2DayPercentChange(
+          data.txCount,
+          oneDayHistory?.txCount ?? 0,
+          twoDayHistory?.txCount ?? 0,
+        );
+
+        const currentLiquidityUSD = data?.totalLiquidity * nativeCurrencyPrice * data?.derivedNativeCurrency;
+        const oldLiquidityUSD =
+          oneDayHistory?.totalLiquidity * nativeCurrencyPriceOld * oneDayHistory?.derivedNativeCurrency;
+
+        // percent changes
+        const priceChangeUSD = getPercentChange(
+          data?.derivedNativeCurrency * nativeCurrencyPrice,
+          oneDayHistory?.derivedNativeCurrency ? oneDayHistory?.derivedNativeCurrency * nativeCurrencyPriceOld : 0,
+        );
+
+        // set data
+        data.priceUSD = data?.derivedNativeCurrency * nativeCurrencyPrice;
+        data.totalLiquidityUSD = currentLiquidityUSD;
+        data.oneDayVolumeUSD = parseFloat(oneDayVolumeUSD);
+        data.volumeChangeUSD = volumeChangeUSD;
+        data.priceChangeUSD = priceChangeUSD;
+        data.liquidityChangeUSD = getPercentChange(currentLiquidityUSD ?? 0, oldLiquidityUSD ?? 0);
+        data.oneDayTxns = oneDayTxns;
+        data.txnChange = txnChange;
+
+        // new tokens
+        if (!oneDayHistory && data) {
+          data.oneDayVolumeUSD = data.tradeVolumeUSD;
+          data.oneDayVolumeNativeCurrency = data.tradeVolume * data.derivedNativeCurrency;
+          data.oneDayTxns = data.txCount;
+        }
+
+        // update name data for
+        updateNameData({
+          token0: data,
+        });
 
-          // HOTFIX for Aave
-          if (data.id === '0x7fc66500c84a76ad7e9c93437bfc5ac33e2ddae9') {
-            const aaveData = await client.query({
-              query: PAIR_DATA('0xdfc14d2af169b0d36c4eff567ada9b2e0cae044f'),
-            });
-            const result = aaveData.data.pairs[0];
-            data.totalLiquidityUSD = parseFloat(result.reserveUSD) / 2;
-            data.liquidityChangeUSD = 0;
-            data.priceChangeUSD = 0;
-          }
-
-          return data;
-        }),
+        // HOTFIX for Aave
+        if (data.id === '0x7fc66500c84a76ad7e9c93437bfc5ac33e2ddae9') {
```

Comment:
Add a comment on what these addresses are ?

Reply:
I think this can be removed, I can't find that token id anywhere on the subgraph. Also, the pair address points to a Uniswap AAVE/WETH pool https://etherscan.io/address/0xdfc14d2af169b0d36c4eff567ada9b2e0cae044f.